File: app/src/main/java/com/nobrain/samples/daumeimagesearch/network/dagger/NetworkModule.java
package com.nobrain.samples.daumeimagesearch.network.dagger;

import com.nobrain.samples.daumeimagesearch.network.SearchApi;
import com.nobrain.samples.daumeimagesearch.network.retrofit.RetrofitCreator;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

@Module
public class NetworkModule {

    @Provides
    public Retrofit provideRetrofit() {
        return RetrofitCreator.createRetrofit();
    }

    @Provides
    public SearchApi provideSearchApi(Retrofit retrofit) {
        return new SearchApi(retrofit);
    }
}

File: app/src/main/java/com/nobrain/samples/daumeimagesearch/network/dagger/NetworkModule.kt
package com.nobrain.samples.daumeimagesearch.network.dagger

import com.nobrain.samples.daumeimagesearch.network.SearchApi
import com.nobrain.samples.daumeimagesearch.network.retrofit.RetrofitCreator

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

@Module
class NetworkModule {

    @Provides
    fun provideRetrofit(): Retrofit = RetrofitCreator.createRetrofit()

    @Provides
    fun provideSearchApi(retrofit: Retrofit) = SearchApi(retrofit)
}
